src/api/SignalClient.ts

```
@@ -356,9 +358,10 @@ export class SignalClient {
               abortSignal?.removeEventListener('abort', abortHandler);
               this.startPingInterval();
               if (resp.message?.case === 'reconnect') {
-                resolve(resp.message?.value);
+                resolve(resp.message.value);
+                this.onReconnectResponse?.(resp.message.value);
               } else {
-                resolve();
+                resolve(undefined);
```

for context to the above comment:

this is line is the culprit. in a reconnect scenario we would resolve a signal reconnect promise with `undefined` instead of the reconnect response, as we consider the first incoming message as a successful signal reconnect.

---

I see, this is handling the case that reconnect is not the most immediate message.

When does that happen? this seems like a rather unexpected behavior.. I think the protocol indicates that Reconnect would need to be the first thing that's sent.. just like JoinResponse.

---

definitely mostly a safe guard. 
if it's unexpected behaviour, we can just throw an error instead of another signal message arrives? 
We'll have to figure out server version support for reconnect response though.

---

yeah, so I'm thinking if we can throw an error if ReconnectResponse comes later than expected. That should trigger a full reconnect sequence.

if the server is sending ReconnectResponse after the initial message, it would be breaking spec. We wouldn't want other clients to implement that behavior.

---

sounds good. Though we cannot reject the promise/throw an error per se after we've resolved the promise already. That will just result in an unhandled exception. 
We could introduce a new error event/callback to handle that. 

Will remove the late response handling parts from the PR for now.

---

cool, sounds good! we can add that separately